partiql-planner/src/main/kotlin/org/partiql/planner/internal/astPasses/NormalizeIdentifier.kt

```diff
@@ -0,0 +1,32 @@
+package org.partiql.planner.internal.astPasses
+
+import org.partiql.ast.Binder
+import org.partiql.ast.Identifier
+import org.partiql.ast.Statement
+import org.partiql.ast.normalize.AstPass
+import org.partiql.ast.util.AstRewriter
+import org.partiql.planner.internal.CaseNormalization
+import org.partiql.planner.internal.IdentifierManager
+
+/**
+ * Case Normalization behavior:
+ *   - UPPERCASE: normalize identifier using uppercase for both lvalue and rvalue
+ *   - LOWERCASE: normalize identifier using lowercase for both lvalue and rvalue
+ *   - EXACTCASE: normalize identifier by preserving case for both lvalue and rvalue
+ *   - NULL: Normalize lvalue by preserving case, no normalization for rvalue.
+ */
+internal class NormalizeIdentifier(
+    caseNormalization: CaseNormalization?
+) : AstPass {
```

Comment:
- Merge this with IdentifierManager
- Use `init` to pick the normalization function ie `(string) -> string`


For example,

```kotlin
internal class NormalizeIdentifier(
    mode: CaseNormalization,
) {

   val normalize: (String) -> String

  init {
      normalize = when (mode) {
          UPPERCASE -> Strings.uppercase
          LOWERCASE -> Strings.lowercase
          EXACTCASE -> { it } 
      }
  }

}
```

Something like that so we aren't null checking and swithcing on mode every time.

Reply:
I guess this suggestion is based off the assumption that we also case preserve (after normalization), which is indeed how this PR is implemented. 

But my thinking is to keep those two separate layer of indirection until at least the RFC is in draft.